### ISSUE TITLE
Applying Self-flow on JIT

### ISSUE THREAD

**Issue by lezhang7** [black-forest-labs/Self-Flow#15]
Thanks authors for sharing the great work. I am recently exploring apply the method on pixel-based JIT model, however I fail to beat the baseline. I was wondering if the authors have tried similar experiments, or is it just I'm making something wrong in my codebase? 

<img width="1439" height="510" alt="Image" src="https://github.com/user-attachments/assets/fde813c8-18e1-4743-906e-8877064b71be" />

**Comment by hila-chefer** [comment]
Hi @lezhang7, thanks for your interest!
We haven’t tried self-flow over JITs, but I don’t see any reason why it shouldn’t work (should be really nice there since there’s even more redundancies).

I can’t really see your full figure there so no idea how many training steps you ran and the exact setup so can’t give a fully formed opinion on this.

**Comment by lezhang7** [comment]
Hi @hila-chefer thanks for your prompt response. We ported Self-Flow to JIT and it comes out FID-neutral against the plain baseline, so we would like to check our setup with you.

### Setup

- **Backbone:** pixel-space flow-matching transformer, 24 blocks, width 1024, patch size 16,
  256×256 → 256 patch tokens. Class-conditional, ImageNet-1k.
- **Training:** AdamW, batch size 1024, base LR 5e-5 scaled by batch/256, 600 epochs, 5 warmup
  epochs, gradient clipping 1.0, timesteps log-normal (P_mean −0.8, P_std 0.8).
- **Evaluation:** FID-50k, Heun, 50 steps, CFG 2.4 on t ∈ [0.1, 1.0].
- **Self-Flow:** dual-timestep scheduling (§3.3) with a per-token velocity target at each token's
  own τ, plus L_rep (§3.4, Eq. 6–7). `R_M = 0.25`, `λ_rep = 0.8`, plain projector + predictor,
  EMA teacher 0.9999 (§A.3), layers `l = 0.3·D → k = 0.7·D` (blocks 7 → 17 at D = 24), and the
  shared noise draw of Eq. 5 (one `x_1` for both views).
- **Convention:** the paper uses `x_t = (1−t)x_0 + t·x_1`, so t = 1 is noise and the teacher takes
  `τ_min = min(t, s)`. Ours is `z = t·x_0 + (1−t)·ε`, so t = 1 is clean and we implemented the
  teacher as `τ_max = max(t, s)` — i.e. the teacher sees the cleaner of the two draws.

### Results

FID vs training epoch. Two independent baseline runs are shown so run-to-run variation is visible.

| epoch | baseline | baseline (2nd run) | Self-Flow |
|---:|---:|---:|---:|
| 40 | 27.17 | 27.74 | 24.81 |
| 80 | 5.20 | 5.18 | 4.33 |
| 200 | 2.89 | 2.93 | **2.67** |
| 400 | 2.53 | 2.56 | 2.46 |
| 520 | 2.53 | 2.51 | 2.53 |
| 600 | 2.70 | 2.42 | — |

Mean FID over epochs 360–560 (the window all three runs share): baseline **2.555**,
baseline 2nd run **2.520**, Self-Flow **2.537**.

Self-Flow lands between the two baseline runs (+0.018 / −0.017), which differ from each other by
0.035. It is ahead for the first half of training and the gap closes over the last third. The
Self-Flow run is at epoch 560 of 600 (2.53); we will update when it finishes.

**Comment by hila-chefer** [comment]
Thanks for the extra info!
One thing I could imagine is, since this is in pixel space, one would need to be a bit more careful about the noise scheduler choice. With RAEs for example, we noticed that the noise scheduler matters a lot more than other latent spaces. 
In cases where redundancies are very significant (eg video), shifting the noise scheduler toward higher noises can be really helpful for DTS to put back more emphasis on the critical higher noise training.
Since pixel space lacks the downsampling factor that a VAE provides, it might be worth it to ablate higher noise regimes when applying DTS in this case specifically.
I hope this helps!

**Comment by lezhang7** [comment]
Thanks, definitely, I am currently reusing shceduler from jit, which has higher noise regimes than dit. Let me try to set higher noise level

**Comment by hanqiuli** [comment]
Hi @lezhang7 any luck with training at higher noise level?
